tests/test_auth.py

```diff
@@ -0,0 +1,59 @@
+import pytest, os
+from unittest import mock
+from auth import load_token_key, Token_gen, logger
+from freezegun import freeze_time
+from datetime import datetime, timedelta
+
+
+
+@pytest.mark.parametrize(
+    ('error', 'path', 'expected_str'),
+    (
+        (FileNotFoundError, 'token_keys.txt', 'FileNotFoundError: no credentials.txt in'),
+        (TypeError, os.path.dirname(os.path.realpath(__file__)) + '/token_key_not_JWK_object.txt', 'is not JWK object')
+    ),
+    ids = ['No token key file case', 'Data in token key file is not JWK object']
+)
+def test_load_token_key_for_errors(error, path, expected_str):
+    with pytest.raises(error):
+        with mock.patch.object(logger, 'error') as mock_error:
+            load_token_key(path)
+    assert expected_str in mock_error.call_args[0][0]
+
+def test_load_key_normal_case():
+    test_token_key = load_token_key(os.path.dirname(os.path.realpath(__file__)) + '/test_token_key.txt')
+    assert test_token_key.export() == '{"k":"foo","kty":"oct"}'
+
+def test_class_Token_create_and_check_token_normal_case():
+    test_token_object  = Token_gen(os.path.dirname(os.path.realpath(__file__)) + '/test_token_key.txt')
+    str_token = test_token_object.create_new_token('testname', 30)
+    assert str_token.count('.') == 2
+    assert test_token_object.check_token(str_token) is True
+
+@pytest.mark.parametrize(
+    ('n_case', 'expected_str'),
+    (
+        (0, 'JWTExpired: token is expired'),
+        (1, 'InvalidJWSSignature: token has invalid signature'),
+        (2, 'ValueError: received string is not JSON Web Token'),
+        (3, 'InvalidJWSObject: invalid token format')
+    ),
+    ids = ['Expired token', 'Token has invalid signature', 'Received string is not JSON Web Token', 'Invalid token format']
+)    
+def test_class_Tokem_check_token_all_error_cases(n_case, expected_str):
```

Comment:
Why not split this test into several, there is very little reuse happening. Alternatively, you can move the token initialisation into the parameters of your test.

Reply:
Done!